python/ray/node.py

```diff
@@ -60,29 +66,60 @@ def __init__(self, ray_params, head=False, shutdown_at_exit=True):
                 os.path.dirname(os.path.abspath(__file__)),
                 "workers/default_worker.py"))
 
+        self._ray_params = ray_params
+
+        self._node_ip_address = ray_params.node_ip_address
+        self._redis_address = ray_params.redis_address
+        self._config = (json.loads(ray_params._internal_config)
+                        if ray_params._internal_config else None)
+
         if head:
             ray_params.update_if_absent(num_redis_shards=1, include_webui=True)
+            self._plasma_store_socket_name = None
+            self._raylet_socket_name = None
+            self._webui_url = None
         else:
-            redis_client = ray.services.create_redis_client(
-                ray_params.redis_address, ray_params.redis_password)
+            self._plasma_store_socket_name = (
+                ray_params.plasma_store_socket_name)
+            self._raylet_socket_name = ray_params.raylet_socket_name
+            redis_client = self.create_redis_client()
+            # TODO(suquark): Replace _webui_url_helper in worker.py in
+            # another PR.
+            _webui_url = redis_client.hmget("webui", "url")[0]
+            self._webui_url = (ray.utils.decode(_webui_url)
+                               if _webui_url is not None else None)
             ray_params.include_java = (
                 ray.services.include_java_from_redis(redis_client))
 
-        self._ray_params = ray_params
-        self._config = (json.loads(ray_params._internal_config)
-                        if ray_params._internal_config else None)
-        self._node_ip_address = ray_params.node_ip_address
-        self._redis_address = ray_params.redis_address
-        self._plasma_store_socket_name = None
-        self._raylet_socket_name = None
-        self._webui_url = None
+        self._init_temp()
 
-        self.start_ray_processes()
+        if not connect_only:
```

Comment:
`connect_only` and `shutdown_at_exit` seem incompatible. Make sure they are not both true.